source/agora/common/crypto/Schnorr.d

```diff
@@ -64,6 +64,481 @@ nothrow @nogc unittest
     assert(verify(kp.V, signature, "Hello world"));
 }
 
+/// Returns: The accumulated values
+/// Note: be aware of accumulators, when we add to a value we must
+/// ensure the value was initialized with an accumulator (not T.init!)
+T combine (T)(T[] elems)
+    if (is(T == Scalar) || is(T == Point))
+{
+    assert(elems.length > 0);
+
+    if (elems.length == 1)
+        return elems[0];
+    else
+        return sum(elems[1 .. $], elems[0]);  // proper accumulator, not T.init!
+}
+
+/// Returns: The accumulated values
+/// Note: be aware of accumulators, when we add to a value we must
+/// ensure the value was initialized with an accumulator (not T.init!)
+T combine (T)(T lhs, T rhs)
+    if (is(T == Scalar) || is(T == Point))
+{
+    assert(lhs != T.init || rhs != T.init);  // at least one must be initialized
+
+    if (lhs == T.init)
+        return rhs;
+    else if (rhs == T.init)
+        return lhs;
+    else
+        return lhs + rhs;
+}
+
+/// validator signature scheme
+unittest
+{
+    import agora.common.BitField;
+    import std.array;
+    import std.algorithm;
+
+    /// example block structure
+    static struct Block
```

Comment:
That looks like a lot of code that shouldn't be in `common/crypto/Schnorr.d`, no?

Reply:
Yes. It's inside a unittest. This PR was meant to be just a PoC to see if the scheme makes sense. 👼